libtenzir/builtins/operators/shell.cpp

```diff
@@ -207,7 +210,7 @@ class shell_operator final : public crtp_operator<shell_operator> {
     {
       // Coroutines require RAII-style exit handling.
       auto at_exit = caf::detail::make_scope_guard([&] {
-        child->close_stdin();
+        child->close_stdinout();
         TENZIR_DEBUG("joining thread");
         thread.join();
       });
```

Comment:
Also, what about the behavior when `shell` is the source?

Reply:
Should be a no-op if I understand the code correctly.